go/common/sgx/pcs/quote.go

```diff
@@ -0,0 +1,584 @@
+	quoteSigSizeLen = 4
+
+	// quoteSigEcdsaP256MinLen is the minimum length of the ECDSA-P256 quote signature in bytes.
+	quoteSigEcdsaP256MinLen = 584
+
+	// ppidDataLen is the PPID certification data length in bytes.
+	ppidDataLen = 404
+)
+
+// Quote is an enclave quote.
+type Quote struct {
+	Header    QuoteHeader
+	ISVReport ReportBody
+	Signature QuoteSignature
+}
+
+// UnmarshalBinary decodes a Quote from a byte array.
+func (q *Quote) UnmarshalBinary(data []byte) error {
+	if len(data) < quoteHeaderLen+reportBodyLen+quoteSigSizeLen {
+		return fmt.Errorf("pcs/quote: invalid quote length")
+	}
+
+	// Quote Header.
+	var offset int
+	if err := q.Header.UnmarshalBinary(data[offset : offset+quoteHeaderLen]); err != nil {
+		return err
+	}
+	offset += quoteHeaderLen
+
+	// ISV Report.
+	if err := q.ISVReport.UnmarshalBinary(data[offset : offset+reportBodyLen]); err != nil {
+		return err
+	}
+	offset += reportBodyLen
+
+	// Quote Signature Length.
+	sigLen := int(binary.LittleEndian.Uint32(data[offset:]))
+	offset += quoteSigSizeLen
+	if len(data) != quoteHeaderLen+reportBodyLen+quoteSigSizeLen+sigLen {
+		return fmt.Errorf("pcs/quote: unexpected trailing data")
+	}
+
+	// Quote Signature.
+	switch q.Header.attestationKeyType {
+	case AttestationKeyECDSA_P256:
+		var qs QuoteSignatureECDSA_P256
+		if err := qs.UnmarshalBinary(data[offset : offset+sigLen]); err != nil {
+			return err
+		}
+		q.Signature = &qs
+	default:
+		return fmt.Errorf("pcs/quote: unsupported attestation key type: %s", q.Header.attestationKeyType)
+	}
+
+	return nil
+}
+
+// Verify verifies the quote.
+//
+// In case of successful verification it returns the TCB level.
+func (q *Quote) Verify(ts time.Time, tcb *TCBBundle) (*TCBLevel, error) {
+	return q.Signature.Verify(&q.Header, &q.ISVReport, ts, tcb)
+}
+
+// QuoteHeader is a quote header.
+type QuoteHeader struct {
+	Version    uint16
+	QESVN      uint16
+	PCESVN     uint16
+	QEVendorID [16]byte
+	UserData   [20]byte
+
+	attestationKeyType AttestationKeyType
+	raw                []byte
+}
+
+// UnmarshalBinary decodes QuoteHeader from a byte array.
+func (qh *QuoteHeader) UnmarshalBinary(data []byte) error {
+	if len(data) != quoteHeaderLen {
+		return fmt.Errorf("pcs/quote: invalid quote header length")
+	}
+
+	qh.Version = binary.LittleEndian.Uint16(data[0:])
+	if qh.Version != 3 {
+		return fmt.Errorf("pcs/quote: unsupported quote version %d", qh.Version)
+	}
+
+	qh.attestationKeyType = AttestationKeyType(binary.LittleEndian.Uint16(data[2:]))
+	qh.QESVN = binary.LittleEndian.Uint16(data[8:])
+	qh.PCESVN = binary.LittleEndian.Uint16(data[10:])
+	copy(qh.QEVendorID[:], data[12:])
+	copy(qh.UserData[:], data[28:])
+
+	qh.raw = data[:quoteHeaderLen]
+
+	return nil
+}
+
+// QEVendorID_Intel is the Quoting Enclave vendor ID for Intel.
+var QEVendorID_Intel = []byte{0x93, 0x9a, 0x72, 0x33, 0xf7, 0x9c, 0x4c, 0xa9, 0x94, 0x0a, 0x0d, 0xb3, 0x95, 0x7f, 0x06, 0x07}
```

Comment:
```suggestion
var QEVendorID_Intel = []byte{0x93, 0x9a, 0x72, 0x33, 0xf7, 0x9c, 0x4c, 0xa9, 0x94, 0x0a, 0x0d, 0xb3, 0x95, 0x7f, 0x06, 0x07} // 939A7233F79C4CA9940A0DB3957F0607
```
which can be searched up to find the quote structure reference doc. I assume that people will be following that while they read this, so there's no need to link the spec, so this is just to help pattern matching.